hepcrawl/spiders/cds_spider.py

```diff
@@ -9,76 +9,133 @@
 
 """Spider for the CERN Document Server OAI-PMH interface"""
 
-from dojson.contrib.marc21.utils import create_record
+from __future__ import absolute_import, division, print_function
+
 from flask.app import Flask
-from harvestingkit.inspire_cds_package.from_cds import CDS2Inspire
-from harvestingkit.bibrecord import (
-    create_record as create_bibrec,
-    record_xml_output,
-)
-from inspire_dojson.hep import hep
-from scrapy import Request
-from scrapy.spider import XMLFeedSpider
+from hepcrawl.spiders.common.oaipmh_spider import OAIPMHSpider
+from inspire_dojson import marcxml2record
+from scrapy import Selector
 
-from . import StatefulSpider
-from ..utils import ParsedItem, strict_kwargs
+from ..utils import (
+    ParsedItem,
+    strict_kwargs,
+)
 
 
-class CDSSpider(StatefulSpider, XMLFeedSpider):
-    """Spider for crawling the CERN Document Server OAI-PMH XML files.
+class CDSSpider(OAIPMHSpider):
+    """Spider for crawling cds.cern.ch OAI-PMH.
 
     Example:
-        Using OAI-PMH XML files::
-
-            $ scrapy crawl \\
-                cds \\
-                -a "source_file=file://$PWD/tests/functional/cds/fixtures/oai_harvested/cds_smoke_records.xml"
+        Using OAI-PMH service::
 
-    It uses `HarvestingKit <https://pypi.python.org/pypi/HarvestingKit>`_ to
-    translate from CDS's MARCXML into INSPIRE Legacy's MARCXML flavor. It then
-    employs `inspire-dojson <https://pypi.python.org/pypi/inspire-dojson>`_ to
-    transform the legacy INSPIRE MARCXML into the new INSPIRE Schema.
+            $ scrapy crawl CDS \\
+                -a "sets=cerncds:hep-th" -a "from_date=2017-12-13"
     """
-
     name = 'CDS'
-    iterator = 'xml'
-    itertag = 'OAI-PMH:record'
-    namespaces = [
-        ('OAI-PMH', 'http://www.openarchives.org/OAI/2.0/'),
-        ('marc', 'http://www.loc.gov/MARC21/slim'),
-    ]
+    source = 'CDS'
 
     @strict_kwargs
-    def __init__(self, source_file=None, **kwargs):
-        super(CDSSpider, self).__init__(**kwargs)
-        self.source_file = source_file
+    def __init__(
+        self,
+        url='http://cds.cern.ch/oai2d',
+        format='marcxml',
```

Comment:
does it really make sense to have a parametric format? the parser understands only MARCXML anyway.

Reply:
Yes, I understand that. What I don't understand is why this is a parameter that is part of the API of the spider. There is no other format that will result in a correct result, because the parser understands only `marcxml` anyway. So it's fake parametricity. It's not too important though, we can always remove it later.